### PR TITLE
Added a checkbox-changed event to checkbox

### PR DESCRIPTION
**Pull request by aleks-elkin**
Added a handler for click event to update the `checked` property status of checkbox.

### CODE REVIEW

File: packages/components/checkbox/src/checkbox.js [hunk 1]
```diff
@@ -26,12 +29,22 @@ export class TSCheckbox extends TSElement {
 		return this.dir || this.bodyDir;
 	}
 
+	onClick(e) {
+		e.stopPropagation();
+		if (this.disabled) {
+			return;
+		}
+		this.checked = !this.checked;
+		this.dispatchCustomEvent(customEventNames.CHECKBOX_CHANGED, this.checked);
```
Comment (esmailpour-hosein) [inline packages/components/checkbox/src/checkbox.js:38]: I think it's a good idea to send the original event object beside the new checked value.

Comment (esmailpour-hosein) [reply packages/components/checkbox/src/checkbox.js:38]: Like this one : https://github.com/Tradeshift/elements/blob/1a3c45c188d00116446f0ce7bb2d14c8983d78a2/packages/components/file-uploader-input/src/file-uploader-input.js#L147

**Comment by aleks-elkin** [comment]
> What do you think if we just let them know they can use MutationObserver? 🤔
> https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver
@esmailpour-hosein 

Well, the general idea looks good, but we still need to fix the component: currently the `checked` attribute is not reflected properly because of behavior of the `checked` attribute of the HTML input element:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#checked

> A Boolean attribute indicating whether or not this checkbox is checked by default (when the page loads). It does not indicate whether this checkbox is currently checked: if the checkbox’s state is changed, this content attribute does not reflect the change.

**Comment by esmailpour-hosein** [comment]
Yes, you are right, I test it out and works fine, but don't know what would be missed in this way.
Other than that this component, missing the light DOM input, which make `name` and `value` attributes redundant. We should take care of that as well in future.

**Comment by aleks-elkin** [comment]
> Yes, you are right, I test it out and works fine, but don't know what would be missed in this way.
> Other than that this component, missing the light DOM input, which make `name` and `value` attributes redundant. We should take care of that as well in future.

Yes, it is one of the tasks for future improvement: add support for forms. But currently developers cannot use it effectively because `checked` status not updated after user interactions, so I wanted to solve this one.

**Pull request merged**